src/main/java/com/gluonhq/richtextarea/ParagraphTile.java

```diff
@@ -145,40 +141,79 @@ TextFlow getTextFlow() {
         return textBackgroundColorPaths;
     }
 
-    // graphicFactoryProperty
-    private final ObjectProperty<BiFunction<Integer, ParagraphDecoration.GraphicType, Node>> graphicFactoryProperty = new SimpleObjectProperty<>(this, "graphicFactory") {
-        @Override
-        protected void invalidated() {
-            graphicBox.getChildren().clear();
-            ParagraphDecoration decoration = paragraph.getDecoration();
-            int indentationLevel = decoration.getIndentationLevel();
-            Node node = get().apply(indentationLevel, decoration.getGraphicType());
-
-            if (node != null) {
-                if (node instanceof Label) {
-                    String text = ((Label) node).getText();
-                    if (text != null && text.contains("#")) {
-                        AtomicInteger ordinal = new AtomicInteger();
-                        viewModel.getParagraphList().stream()
-                                .filter(p -> p.getDecoration().getIndentationLevel() == indentationLevel)
-                                .peek(p -> {
-                                    if (p.getDecoration().getGraphicType() == ParagraphDecoration.GraphicType.BULLETED_LIST) {
-                                        // restart ordinal if previous paragraph with same indentation is a bulleted list
-                                        ordinal.set(0);
-                                    } else {
-                                        ordinal.incrementAndGet();
-                                    }
-                                })
-                                .filter(p -> paragraph.equals(p))
-                                .findFirst()
-                                .ifPresent(p ->
-                                        ((Label) node).setText(text.replace("#", "" + ordinal.get())));
-                    }
+    private void updateBox(BiFunction<Integer, ParagraphDecoration.GraphicType, Node> graphicFactory) {
+        graphicBox.getChildren().clear();
+
+        ParagraphDecoration decoration = paragraph.getDecoration();
+        int indentationLevel = decoration.getIndentationLevel();
+        Node graphicNode = null;
+        if (graphicFactory != null) {
+            graphicNode = graphicFactory.apply(indentationLevel, decoration.getGraphicType());
+        }
+        double spanPrefWidth = Math.max((indentationLevel - (graphicNode != null ? 1 : 0)) * INDENT_PADDING, 0d);
```

Comment:
I would reverse the condition to `graphicNode == null ? 0 : 1`

Reply:
done